### PR TITLE
Cleanup rewrite

### LINKED ISSUE #3: Review the best way to improve testing

### LINKED ISSUE #5: add a better pr template
```
# Introduction

# Linked Issues

# Dependencies

# Test
- [ ] Tested Locally
- [ ] Documented

# Review
- [ ] Ready for review
- [ ] Ready to merge

# Deployment

```

### PR DESCRIPTION
**Pull request by ncatelli**
# Introduction
The API compliant rewrite of paste-click's backend to resolve a lot of testing and development problems.
# Linked Issues
resolves #3 
resolves #5 

# Dependencies

# TODO
- [x] Update documentation
- [ ] Manual testing from rest of team

# Test
- [x] Tested Locally
- [x] Documented

# Review
- [x] Ready for review
- [x] Ready to merge

# Deployment

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -21,16 +19,15 @@ ENV SERVICE_USER "paste-click"
...
+    adduser -D -G ${SERVICE_USER} ${SERVICE_USER}
 
 COPY --from=builder /${APP_NAME} /opt/${APP_NAME}/bin/
 
 RUN chown -R ${SERVICE_USER}:${SERVICE_USER} /opt/${APP_NAME} && \
     chmod +x /opt/${APP_NAME}/bin/${APP_NAME}
 
 WORKDIR "/opt/$APP_NAME/"
-USER ${SERVICE_USER}
+#USER ${SERVICE_USER}
```
Comment (RaasAhsan) [inline Dockerfile]: This need to be around for any reason ?

File: main.go [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	"github.com/PacketFire/paste-click/lib/objectstore/drivers/mock"
+
+	"github.com/gorilla/mux"
+)
+
+// store takes a string representing drivers and attempts to return a
+// corresponding driver if there is no match, nil is returned.
+func store(driverName string) objectstore.ObjectStore {
+	switch driverName {
+	case `local`:
```
Comment (RaasAhsan) [inline main.go]: Would it be better to call this `fs` or `filesystem`, since you could point it to a NFS mounted directory if you wanted?

**Comment by ncatelli** [comment]
May need to pull https://github.com/rakyll/magicmime back in it looks like pulling the post bodies mimetype from the Content-Type header is unreliable.

File: main.go (hunk 2 above)
Comment (ncatelli) [reply main.go]: I agree, this has been renamed to `fs`.

**Pull request merged**